packages/flutter_tools/lib/src/test/runner.dart

```diff
@@ -54,13 +55,7 @@ Future<int> runTests(
     testArgs.addAll(<String>['-r', 'compact']);
   }
 
-  if (enableObservatory) { // (In particular, for collecting code coverage.)
-    // Turn on concurrency, but just barely. This is a trade-off between running
-    // too many tests such that they all time out, and too few tests such that
-    // the tests overall take too much time. The current number is empirically
-    // based on what our infrastructure can handle, which isn't ideal...
-    testArgs.add('--concurrency=2');
-  }
+  testArgs.add('--concurrency=$concurrency');
```

Comment:
Before, it would default to N/2 for regular tests, and 2 for coverage, where N is the number of CPUs. But that was based on what we needed for Travis. Now it defaults to N-2 for everything.

I'm not sure what the defaults should actually be. N/2 is definitely too low for regular tests when you have a lot of CPUs (on my 20 CPU machine, about half the machine, around 8 CPUs, is entirely idle when I'm running tests). I don't know what Cirrus can take. So I figured that scaling the numbers back up and seeing what would happen was probably the next move. :-)